src/components/Portal/Portal.js

```diff
@@ -0,0 +1,44 @@
+import { PureComponent }             from 'react'
+import { createPortal, findDOMNode } from 'react-dom'
+import PropTypes                     from 'prop-types'
+
+class Portal extends PureComponent {
+  static propTypes = {
+    children: PropTypes.node.isRequired,
+    container: PropTypes.oneOfType([
+      PropTypes.node,
+      PropTypes.func
+    ])
+  }
+
+  constructor(props) {
+    super(props)
+    this.setContainer(this.props.container)
+  }
+
+  componentWillReceiveProps(nextProps) {
```

Comment:
Can we refactor this to avoid `componentWillReceiveProps` since it's getting deprecated?